test/e2e/operator_test.go

```diff
@@ -916,20 +916,214 @@ func TestIngressGroup(t *testing.T) {
 
 	firstAddress, err := getIngress(context.TODO(), t, kubeClient, defaultTimeout, ingName1)
 	if err != nil {
-		t.Fatalf("did not get expected available condition for ingress: %v", err)
+		t.Fatalf("did not get load balancer address for ingress: %v", err)
 	}
 
 	for _, ingressName := range []types.NamespacedName{ingName2, ingName3} {
 		address, err := getIngress(context.TODO(), t, kubeClient, defaultTimeout, ingressName)
 		if err != nil {
-			t.Fatalf("did not get expected available condition for ingress %s: %v", ingressName, err)
+			t.Fatalf("did not get load balancer address for ingress %s: %v", ingressName, err)
 		}
 		if address != firstAddress {
 			t.Errorf("ingress %s does not have the address %s, instead has %s", ingressName, firstAddress, address)
 		}
 	}
 }
 
+// TestAWSLoadBalancerControllerWithDefaultLoadBalancerClass tests the basic happy flow for NLB.
+// "service.k8s.aws/nlb" load balancer class is used as the default for
+// the service reconciliation done by aws-load-balancer-controller.
+func TestAWSLoadBalancerControllerWithDefaultLoadBalancerClass(t *testing.T) {
```

Comment:
> If I remove "service.k8s.aws/nlb" it defaults to a classic load balancer.

If `service.k8s.aws/nlb` load balancer class is not set on a service - the in-tree lb controller will be used instead of the aws-load-balancer-controller.

I tried to make a test scenario similar to the existing `TestAWSLoadBalancerControllerWithDefaultIngressClass` which uses `alb` as the ingress class. It's [the default ingress class set in the API](https://github.com/openshift/aws-load-balancer-operator/blob/main/api/v1/awsloadbalancercontroller_types.go#L85).
There is [a plan](https://issues.redhat.com/browse/NE-1344) to add a new field for the load balancer class to the API. `service.k8s.aws/nlb` will be its default value.

Reply:
I think I understand. Are you saying that a new knob for the load balancer class would allow for the `service.beta.kubernetes.io/aws-load-balancer-type` to be used instead of service's `LoadBalancerClass`? 

Otherwise, what's the point of adding an API for load balancer class in ALBO when you have to use  `service.spec.loadBalancerClass` anyways?

Reply:
The aws-load-balancer-controller reconciles only the services of type `LoadBalancer`. The thing is that if a service of type LoadBalancer is created without any load balancer class the in-tree Kubernetes controller has to take care of the provisioning of the AWS LB. So there came a convention that if a service needs to be taken care by out-of-tree controller (=aws-load-balancer-controller), thi service needs to set a pre-defined LB class: `service.k8s.aws/nlb`. 

The new know would allow the aws-load-balancer-controller to reconcile services of the type LoadBalancer of a any given LB class, not only the default (`service.k8s.aws/nlb`).

Reply:
Ah okay I understand now. You can change the `service.k8s.aws/nlb` load balancer class to be something else and ALBC will provision it.